python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -218,21 +221,69 @@ def run(
 
         self.log.info("Measuring %s", metricName)
 
+        # Validation to require that gri bands are in the input set
+        if set(currentBands).issuperset(set(['g', 'r', 'i'])) is False:
+            self.log.warn("Data in gri bands required for wPerp calculation.")
+            return Struct(measurement=Measurement(metricName, np.nan * u.mmag))
+
         # If accessing objectTable_tract, we need to append the band name at
         #   the beginning of the column name.
         if currentBands is not None:
             prependString = currentBands
         else:
             prependString = None
 
+        # import pdb; pdb.set_trace()
+
         # filter catalog
         catalog = selectors.applySelectors(catalog,
                                            self.config.selectorActions,
                                            currentBands=currentBands)
 
-        extVals = extinction_corr(catalog, currentBands)
-
-        # Next, figure out how to call Sophie's fit function(s).
-        # The main issue is figuring out how to give the bands in the order we need?
-        fitParams = stellarLocusFit(xs, ys, self.config.stellarLocusFitDict)
+        # Filter based on configured r-mag limits:
+        # rmag_tmp = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value
+        # magfilter = (rmag_tmp < self.config.faint_rmag_cut) &\
+        #             (rmag_tmp > self.config.bright_rmag_cut)
+        # catalog = catalog[magfilter]
+
+        # Create a SkyCoord object to get the extinction:
+        sc = SkyCoord(catalog['coord_ra']*u.deg, catalog['coord_dec']*u.deg)
+
+        extVals = extinctionCorrTable(sc, currentBands)
+
+        g0 = (catalog.g_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_g']
+        r0 = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_r']
+        i0 = (catalog.i_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_i']
+
+        # Call the stellar locus fit function:
+        p1vals, p2vals, fitParams = stellarLocusFit(g0-r0, r0-i0,
+                                                    self.config.stellarLocusFitDict)
+
+        # Ivezic+2004 defines wPerp between -0.2 < p1 < 0.6.
+        # Also, do some expanded color cuts to get rid of large color outliers that
+        #   skew the rms.
+        magcushion = 0.2
+        okp1 = (p1vals > -0.2) & (p1vals < 0.6) &\
+               ((r0-i0) < fitParams['yMax']+magcushion) &\
+               ((r0-i0) > fitParams['yMin']-magcushion) &\
+               ((g0-r0) < fitParams['xMax']+magcushion) &\
+               ((g0-r0) > fitParams['xMin']-magcushion)
+
+        # import pdb; pdb.set_trace()
+
+        if np.size(p2vals[okp1[0]]) > 2:
+            p2_rms = calcQuartileClippedStats(p2vals[okp1[0]]).rms * u.mag
```

Comment:
There are a few PEP8 violations here and throughout.

Reply:
I found none in the final commit, so you must be referring to an earlier commit.